src/facade/async.ts

```diff
@@ -103,20 +105,63 @@ export class EventEmitter<T> {
     this._isAsync = isAsync;
   }
 
-  private _dispose( cbRef ) {
+  private _dispose( cbRef: EventHandler<T> ) {
     EventEmitter.createDisposable( cbRef, this );
   }
 
   /** @internal */
   wrapNgExpBindingToEmitter( cb: Function ) {
-    this._ngExpressionBindingCb = cb;
+    //used in reassignBindingsAndCreteEventEmitters to attach the original @Output binding to the instance new EventEmitter
+    this._ngExpressionBindingCb = ( cb as _ngEmitPayloadEvent<T> );
   }
 
   emit( value: T ) {
     EventEmitter.createEmit( value, this );
   }
 
-  subscribe( generatorOrNext?: Function, error?: any, complete?: any ): Function {
+  subscribe( generatorOrNext?: EventHandler<T>, error?: any, complete?: any ): Function {
     return EventEmitter.createSubscribe( generatorOrNext, this );
   }
 }
+
+
+/**
+ * Use by parent component to mark a function as an handler of a custom Event
+ * 
+ *  ### Examples
+ * 
+ *  In the following esample, `ZParent` (parent component) handle `open` and `close` events emitted by `Zippy` (child component)
+ * 
+ * ```
+ * @Component({
+ *   selector: 'z-parent',
+ *   template: `
+ *   Content is visible? {{ $ctrl.zippyState }}
+ *   <zippy open="$ctrl.showState($event)" close="$ctrl.showState($event)">
+ *      Alternate <i>show</i> and <i>hide</i> of this <b>content</b> ... 
+ *   </zippy>
+ * `})
+ * export class ZParent {
+ *   public zippyState: boolean = true;
+ *   public showState: EventHandler<boolean> = ( visible ) => {
+ *      this.zippyState = visible; // HERE visibile is a boolean === $event passed from zippy child component events
+ *   }  
+ * }
+ * ```
+ */
+export interface EventHandler<T> extends Function {
+  ($event: T): void;
+}
+
+/** @internal */
+interface _ngEmitPayloadEvent<T> extends Function {
```

Comment:
it is not recommended to prefix Interfaces or Type aliases with underscores. if it's private, it's not exported, simple as that :)